src/GraphQL.Client/UriExtensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+
+namespace GraphQL.Client.Http
+{
+    public static class UriExtensions
+    {
+        /// <summary>
+        /// Returns true if <see cref="Uri.Scheme"/> equals "wss" or "ws"
+        /// </summary>
+        /// <param name="uri"></param>
+        /// <returns></returns>
+        public static bool HasWebSocketScheme(this Uri uri) => uri.Scheme.Equals("wss") || uri.Scheme.Equals("ws");
```

Comment:
This should be a case-insensitive comparison since `"HTTPS"` is just as valid as `"https"` for the scheme.